### PR TITLE
Dart Methods (New Pull Request)

### LINKED ISSUE #4252: [Concept Entry] Methods: New concept entry `methods.md`
### Reason/inspiration (optional)

We would like a new term entry on the `Methods` concept for [Dart](https://www.codecademy.com/resources/docs/dart). The entry should go under the following path `docs/content/dart/concepts/methods/methods.md`.

The entry should describe the term and how it is used within Dart. Also include: 
- There should be "## Syntax" and "## Example" headings in this entry.
- Explain briefly what methods are in Dart with suitable code snippets.

### Entry Type (select all that apply)

New Concept Entry for Existing Topic

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by brandk94**
### Description

- Created a new sub-directory called 'methods' for the existing directory:   'content/dart/concepts'
- Created and edited a new file called 'methods.md' for the directory:    'content/dart/concepts/methods'

### Issue Solved

Closes #4252 

### Type of Change

- Adding a new entry
- Editing an existing entry (fixing a typo, bug, issues, etc)
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

**Comment by brandk94** [comment]
Hi @cigar-galaxy82 and @ishg-153. I looked at the results of the commit push test and saw that the "ci (format:verify)" and "ci (test)" workflows failed.  I looked to see what caused the failure, and for the "ci (test)" workflow, it turns out there was a spelling error with the "CatalogContent" metadata in the Dart methods.md file that I made.  The word "Catalog" in "CatalogContent" was misspelled "Catelog", but now I fixed it to "Catalog".  If there any more errors with this, please let me know. Thank you.

**Comment by ishg-153** [comment]
@cigar-galaxy82 Please are you working on reviewing this PR?

**Comment by cigar-galaxy82** [comment]
Change the structure of the of the entry as mentioned below

Syntax section
Return Value section
Parameters section

Then have a types of methods section, in this mention all the types shown below

Instance Method 
Static Method
Short-Hand Method

File: content/dart/concepts/methods/methods.md [hunk 1]
```diff
@@ -0,0 +1,585 @@
...
+  <return_type> shortHandMethod(<parameters>) => // Code is inserted here all on a single line after the arrow
+
+}
+```
+
+## Example
+
+A method named `.helloWorld()` that prints the statement `"Hello World"` is defined in a `Hello` class:
+
+```dart
```
Comment (cigar-galaxy82) [inline content/dart/concepts/methods/methods.md:103]:
```suggestion
```pseudo
```

**Review by cigar-galaxy82 — CHANGES_REQUESTED** [review]

**Comment by cigar-galaxy82** [comment]
Hi @brandk94 thank you for contributing to Codecademy docs, I have suggested changes please modify. Also please look at the structure of the entry is have suggested

**Comment by brandk94** [comment]
Hi @cigar-galaxy82. I made significant changes to my Dart methods.md file in accordance with the suggested changes you requested, and some other changes to make the file context look nice and consistent. Please let me know what you think and if any further changes need to be made. Thank you.

**Review by cigar-galaxy82 — APPROVED** [review]

**Review by Sriparno08 — APPROVED** [review]
Looks good, @brandk94!

**Pull request merged**